### DISCUSSION TITLE
OCR, Braille Status 처리 관련 질문

### DISCUSSION THREAD

**Discussion by jun-brro** [sunnybraille/sunnybraille-server#95] [Q&A] [answered]
<img width="657" alt="image" src="https://github.com/sunnybraille/sunnybraille-server/assets/115399447/4f3ab21f-35dc-4f23-bf9b-4556a5cf1119">

현재 클라이언트 진행사항을 보았을 때, 질문 2가지 있습니다.

1. OCR 처리 percent는 0 or 100 이렇게만 주어지나요? 만약 그렇다면 Progress Bar 보다는 체크형태로, 진행 완료되었다고 표시해주는게 적절할 듯 해서요..!@gitchannn 
2. 혹시 Braille process를 퍼센트화 하기 위한 방안이 있을까요? @J-nowcow @seojunsoo 

<img width="601" alt="image" src="https://github.com/sunnybraille/sunnybraille-server/assets/115399447/940daa8a-a5a1-4dd4-a8f8-42ee3f9580d5">
<img width="413" alt="image" src="https://github.com/sunnybraille/sunnybraille-server/assets/115399447/6d13c5ab-05d7-4ca8-871a-747f06da6cbb">

**Comment by gitchannn** [comment, marked as answer]
일단 지금 상태에서는 OCR API는 비동기적으로 구현되어 있지 않아서 서버에서 점역 서버로 요청을 보낸 뒤에 기다리다가 결과값을 받아오고 있습니다.
동기적으로 기다리기 때문에 중간에 서버 쪽에서 다른 요청을 보낼 수가 없고 따라서 현재 상황으로는 백엔드 서버 측에서도 O X로만 결과를 알 수 있는 상황입니다.
비동기를 구현하기 위해서는 저도 아직 구현해본 적은 없지만 `롱폴링`, `소켓` 등을 사용하면 된다고 알고 있습니다.
작은 편의성 위해서 너무 큰 품이 들 것 같아서, 개인적인 의견으로는 현재로는 백엔드에서 정상 처리 되었는지만을 0 또는 100으로 반환하고, 클라이언트에서는 사용자에게 체크 표시 정도를 통해 표현하면 충분할 것 같다고 생각합니다.

**Reply by jun-brro** [reply]
넵 그럼 체크표시 활성화 방안으로 변경하겠습니다.
혹시 저번에 말씀주셨던 brailleTranslationPercentDone 100으로 바꾸는 것은 혹시 처리가 된 것이 맞을까요? 계속 이 화면에서 멈춰있어서, 한번 확인 부탁드립니다..!
<img width="318" alt="image" src="https://github.com/sunnybraille/sunnybraille-server/assets/115399447/5901ab48-5975-4967-a119-2fca3d7fe70f">

**Reply by gitchannn** [reply]
braille percent 바꾸는것까지 반영 완료했습니다! 
포스트맨으로 정상 작동하는 것도 확인 완료우

https://github.com/sunnybraille/sunnybraille-server/pull/100
